cmake/compiler/gcc/compiler_flags.cmake

```diff
@@ -166,6 +166,7 @@ set_compiler_property(PROPERTY freestanding -ffreestanding)
 
 # Flag to enable debugging
 set_compiler_property(PROPERTY debug -g)
+set_compiler_property(PROPERTY debug -gdwarf-4)
```

Comment:
Although dwarf-4 standard is from 2010 should we use a `check_set_compiler_property()` to ensure that we are not breaking things for proprietary compilers that are based on an old gcc ?

Reply:
Updated to use `check_set_compiler_property()`.

According to code, pyelftools supports version 2 to 4.

Reply:
Thanks for checking.